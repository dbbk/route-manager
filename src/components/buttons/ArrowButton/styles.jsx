// @flow

import styled from 'styled-components';
import { prop } from 'styled-tools';
import BackArrow from '../../icons/ui/BackArrow';

export const StyledBackArrow = styled(BackArrow)`
  color: ${prop('theme.colors.brand.primary')};
  transform: rotate(180deg);
`;
