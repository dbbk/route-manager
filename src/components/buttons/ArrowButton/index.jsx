// Lib
// -------------
import React from 'react';

// Components
// -------------

// Styles
// -------------
import { StyledBackArrow } from './styles';

// Flow Types
// -------------
type Props = {
  element?: any,
};

export default function ArrowButton({ element: El, ...props }: Props) {
  return (
    <El {...props}>
      <StyledBackArrow width="18" height="18" />
    </El>
  );
}
