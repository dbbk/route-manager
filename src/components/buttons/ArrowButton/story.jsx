// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import ArrowButton from './';
import Anchor from '../../text/Anchor';

storiesOf('Components / Buttons / ArrowButton', module).add('default', () => (
  <ArrowButton element={Anchor} to="https://www.google.com" external />
));
