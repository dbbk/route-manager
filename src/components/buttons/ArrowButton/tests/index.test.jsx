// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import ArrowButton from '../';

function El({ children, to }: { children?: any, to: string }) {
  return <abbr title={to}>{children}</abbr>;
}

describe('<ArrowButton />', () => {
  test('with initial props', () => {
    const tree = shallow(<ArrowButton href="/magazine" external />);
    expect(tree).toMatchSnapshot();
  });

  test('with custom element', () => {
    const tree = mount(
      <ArrowButton element={El} to="https://www.google.com" />,
    );
    expect(tree).toMatchSnapshot();
  });
});
