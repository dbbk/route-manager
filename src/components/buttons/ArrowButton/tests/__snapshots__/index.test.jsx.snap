// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowButton /> with custom element 1`] = `
.c0 {
  cursor: pointer;
  text-decoration: none;
}

.c2 {
  fill: currentColor;
}

.c1 {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

<ArrowButton
  element={[Function]}
  to="https://www.google.com"
>
  <Anchor
    element={[Function]}
    to="https://www.google.com"
  >
    <Styled(AnchorComponent)
      element={[Function]}
      target="_blank"
      to="https://www.google.com"
      variant="standard"
    >
      <AnchorComponent
        className="c0"
        element={[Function]}
        target="_blank"
        to="https://www.google.com"
        variant="standard"
      >
        <El
          className="c0"
          element={[Function]}
          target="_blank"
          to="https://www.google.com"
          variant="standard"
        >
          <abbr
            title="https://www.google.com"
          >
            <Styled(Styled(BackArrow))
              height="18"
              width="18"
            >
              <Styled(BackArrow)
                className="c1"
                height="18"
                width="18"
              >
                <BackArrow
                  className="c1 c2"
                  height="18"
                  width="18"
                >
                  <svg
                    className="c1 c2"
                    height="18"
                    viewBox="0 0 20 20"
                    width="18"
                  >
                    <path
                      d="M1.71 10L10 1.71l.71.7L3.62 9.5H18.5v1H3.62l7.09 7.09-.71.7L1.71 10z"
                    />
                    <path
                      d="M18 10H2.41l1.71 1.71L10 17.59 2.41 10H18m-8-9l-9 9 9 9 1.41-1.41L4.83 11H19V9H4.83l6.59-6.59L10 1z"
                    />
                  </svg>
                </BackArrow>
              </Styled(BackArrow)>
            </Styled(Styled(BackArrow))>
          </abbr>
        </El>
      </AnchorComponent>
    </Styled(AnchorComponent)>
  </Anchor>
</ArrowButton>
`;

exports[`<ArrowButton /> with initial props 1`] = `
<Anchor
  external={true}
  href="/magazine"
>
  <Styled(Styled(BackArrow))
    height="18"
    width="18"
  />
</Anchor>
`;
